packages/vite-plugin-cloudflare/src/cloudflare-environment.ts

```diff
@@ -178,7 +156,29 @@ export function createCloudflareEnvironmentOptions(
 					external: [...cloudflareBuiltInModules],
 				},
 			},
-			webCompatible: true,
+			optimizeDeps: {
+				// Note: ssr pre-bundling is opt-in, and we need to enabled it by setting noDiscovery to false
+				noDiscovery: false,
+				exclude: [
+					...cloudflareBuiltInModules,
+					...builtinModules.concat(builtinModules.map((m) => `node:${m}`)),
+				],
+				esbuildOptions: {
+					resolveExtensions: [
+						'.mjs',
+						'.js',
+						'.mts',
+						'.ts',
+						'.jsx',
+						'.tsx',
+						'.json',
+						'.cjs',
+						'.cts',
+						'.ctx',
+					],
+				},
+			},
+			keepProcessEnv: true,
```

Comment:
as far as I understand `keepProcessEnv: true` basically asks Vite not to mess with `process.env` which I think is something that we want ([source](https://github.com/vitejs/vite/blob/ce0eec91c9b5a91c52bf5b9bae6173d35f6e1649/packages/vite/src/node/config.ts#L246-L250))

Reply:
Interesting. Perhaps using `false` so that it replaces it with an empty object would be good when `nodejs_compat` is not enabled?

Reply:
I don't have a super strong opinion on this one

I think I am a bit more inclined to say that `keepProcessEnv: true` is always the best in our case so that we completely leave the handling/initialization of `process.env` up to workerd

but again I am fine with either 🙂  (it should also be simple to revert this back later, although having to check if nodejs_compat is set is sound to force us to move this code around unfortunately I think (as I did here: https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/52/files#diff-0217c39a7be6fcdb410ac8dfea2ae2998b882a1aaba3f2e83ff2831a0beee7bfR71-R73))

Reply:
I agree that keeping it set to true for now seems the best approach.

Reply:
ah ok, let's go with that for now then (and we can always revisit later on if needed) 🙂